test/generatortests.js

```diff
@@ -14,7 +15,7 @@ var runGeneratorTests = function(testcases) {
                 // It seems reasonable to expect that the generated xpath
                 // should parse back to the same object, although this may 
                 // not always hold true.
-                equal(parsed.toString(), xpath.parse(parsed.toXPath()).toString(), "" + i + " produced same result when reparsed.");
+                equal(parsed.toString(), xpath.parse(parsed.toHashtag()).toString(), "" + i + " produced same result when reparsed.");
```

Comment:
Agreed. It made more sense in my old implementation. https://github.com/dimagi/js-xpath/commit/ebef5c0c6f904e2e54f9823890e075c7ea76128d